ipa-core/src/protocol/context/batcher.rs

```diff
@@ -152,10 +158,16 @@ impl<'a, B> Batcher<'a, B> {
         let total_count = min(self.records_per_batch, remaining_records);
         let record_offset_in_batch = usize::from(record_id) - first_record_in_batch;
         let batch = self.get_batch_by_offset(batch_offset);
-        assert!(
-            !batch.pending_records[record_offset_in_batch],
-            "validate_record called twice for record {record_id}",
-        );
+        if batch.pending_records.len() <= record_offset_in_batch {
+            batch
```

Comment:
I think it is worth writing a test for this - it is important that we can resize this vec if we get over 50M (we definitely do in PRF)

Reply:
After thinking about it, maybe we don't want to take this change at all? I don't think it's much better than `1 << 30`, even with a test. What we really want is for the protocols to specify an appropriate batch size.

Reply:
Maybe. Protocols should be able to correctly estimate the number of records, compared to number of multiplications. It could be also true that we are ok giving them some slack.

We could start without resizing it and bring it back if needed

Reply:
I added a test. I added a smaller value of `TARGET_PROOF_SIZE` for `cfg(test)`. This test case would be fine with the default, but I think it's valuable to cover the multi-batch case for batched aggregation and for other protocols.